Add audit log acceptance test

koevskinikola opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->
Add acceptance tests for the audit log feature.

- The tests in this PR will cover the creation of audit logs for Process Instance Modification operations.
- They will test the Camunda Client audit log search command to assert the creation of audit logs from multiple PI Modification commands.
- They will test the Camunda Client audit log get command to assert the successful retrieval of a single audit log entry after it becomes available in secondary storage.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes #41129

koevskinikola committed:
test: add audit log acceptance test

koevskinikola committed:
test: add coverage for audit log get endpoint

koevskinikola committed:
test: audit logs are isolated by tenant

koevskinikola committed:
test: add coverage for composite permissions and audit log categories

koevskinikola commented on qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogIT.java:
```diff
@@ -0,0 +1,481 @@
+          .untilAsserted(
+              () -> {
+                // First, search for the audit log entry to get its ID
+                final var auditLogItems = adminClient.newAuditLogSearchRequest().send().join();
```
This test covers the retrieval of a single audit log entry by the `auditLogKey`. We only need the search request to retrieve at least on audit log entry, so that we can obtain an `auditLogKey` that will be used by the GET request.

It's not relevant what type of request it is, although we can make it more robust in the future, to avoid flakiness.

tanjinyi commented on qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogIT.java:
```diff
@@ -0,0 +1,481 @@
+            .withKey(processInstanceKey)
+            .exists();
+    if (piCreated) {
+      Awaitility.await("Audit log entry is created for process instance modification")
```
```suggestion
      Awaitility.await("Audit log entry is created for process instance creation")
```

tanjinyi commented on qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogIT.java:
```diff
@@ -0,0 +1,481 @@
+                      .join();
+
+              // Admin should see both audit logs
+              assertThat(adminAuditLogs.items()).isNotEmpty();
```
should probably strengthen the assertion here to check for >=2 items

tanjinyi reviewed (approved):
Preapproving the PR with some minor comments.

koevskinikola committed:
test: add test coverage for grpc api and harden tests

Ensure that operations triggered through the
gRPC and REST APIs behave the same in the
context of the audit log.

Add assertions for entity types.

Ensure that expected audit log items assertions
are more precise.

koevskinikola commented on qa/acceptance-tests/src/test/java/io/camunda/it/auditlog/AuditLogIT.java:
```diff
@@ -0,0 +1,481 @@
+                final var auditLog = auditLogItems.items().get(0);
+
+                assertThat(auditLog).isNotNull();
+                assertThat(auditLog.getOperationType()).isEqualTo(AuditLogOperationTypeEnum.CREATE);
```
💡 Good point!
